src/entity/user/experiences/UserExperience.ts

```diff
@@ -1,64 +1,80 @@
 import {
   Column,
   Entity,
+  Index,
   JoinColumn,
-  JoinTable,
-  ManyToMany,
   ManyToOne,
   PrimaryGeneratedColumn,
   TableInheritance,
 } from 'typeorm';
 import type { User } from '../User';
-import type { UserSkill } from '../UserSkill';
-import { ExperienceStatus, UserExperienceType } from './types';
+import { UserExperienceType } from './types';
+import type { Company } from '../../Company';
+import { LocationType } from '@dailydotdev/schema';
+import { listAllProtoEnumValues } from '../../../common';
+import type { DatasetLocation } from '../../dataset/DatasetLocation';
+
+const defaultLocationTypes = listAllProtoEnumValues(LocationType);
 
 @Entity()
 @TableInheritance({ column: { type: 'text', name: 'type' } })
 export class UserExperience {
```

Comment:
Commenting on main entity so not to replicate on each child entity. Could we maybe name the child entities `UserExperience{Child}`? So that we retain the entity name first, and they're all "grouped" together in a list.
So
- UserExperience
- UserExperienceCertification
- UserExperienceEducation
- etc etc

I know they're technically in a `experience` directory, but imho it's neater. I've applied same naming on opportunities `OpportunityJob`